backend/benefit/applications/tests/test_application_search.py

```diff
@@ -64,25 +88,40 @@ def test_search_with_no_search_query(handler_api_client):
 def test_search_with_results(
     handler_api_client, q, detected_pattern, archived, application
 ):
-    # Setup some bogus data
-    application.company.company_name = "Pitkänen Ruuskanen Oyj"
-    application.company.save()
-
-    application.employee.social_security_number = "040337W935P"
-    application.employee.encrypted_social_security_number = "040337W935P"
-    application.employee.first_name = "Mikro Tietokoneinen"
-    application.employee.last_name = "Matriisi-Artikkeli"
-    application.employee.save()
-
-    application.application_number = 125010
-    application.ahjo_case_id = "HEL 2024-000123"
-    application.archived = archived
-    application.save()
-
+    application = setup_application_data(application, archived)
     response = handler_api_client.get(f"/v1/search/?q={q}")
     data = response.json()
 
     assert response.status_code == 200
     assert len(data["matches"]) > 0
     assert data["detected_pattern"] == detected_pattern
     assert data["matches"][0]["application_number"] == application.application_number
+
+
+@pytest.mark.parametrize(
+    "q, detected_pattern",
+    [
+        ("R001", SearchPattern.ARCHIVAL),
+        ("Pitkänen Ruuskanen Oyj", SearchPattern.COMPANY),
+    ],
+)
+def test_search_archival_application(handler_api_client, q, detected_pattern):
+    ImportArchivalApplicationsTestUtility.create_companies_for_archival_applications()
+    call_command("import_archival_applications", filename="test.xlsx", production=True)
+
+    # Make sure import is ok
+    archival_applications = ArchivalApplication.objects.all()
+    assert archival_applications.count() == 2
+
+    searched_app = ArchivalApplication.objects.filter(application_number=q).first()
+    response = handler_api_client.get(f"/v1/search/?q={q}&archival=1")
```

Comment:
This (and the existing call in the previous test) could be reverse URLs like in most tests instead of hardcoded URLs.